go/vt/vttablet/onlineddl/vrepl.go

```diff
@@ -301,19 +304,19 @@ func (v *VRepl) generateFilterQuery() error {
 			if targetCol.IsTextual() && toCollation == collations.Unknown {
 				return vterrors.Errorf(vtrpcpb.Code_INVALID_ARGUMENT, "Character set %s not supported for column %s", targetCol.Charset(), targetCol.Name())
 			}
-
 			if trivialCharset(fromCollation) && trivialCharset(toCollation) && targetCol.Type() != "json" {
 				sb.WriteString(escapeName(name))
+			} else if fromCollation == toCollation && targetCol.Type() != "json" {
```

Comment:
We don't want `&& targetCol.Type() != "json"` here, do we? We already handle the non-JSON to JSON case above.

Reply:
If the target column is NOT a `JSON` type then avoiding the `CONVERT` — if we can — would eliminate the potential case where the target column is used in the table's PK or PKE that is chosen when applying the change. Otherwise we could run into the same kind of stalls mentioned in the noted bug this is fixing on the target/`vplayer` side.

Reply:
In any event, I don't think this is a major issue as the primary issue we've seen on the target/`vplayer` side is where we were unable to use the desired index because of the `CONVERT` usage and you can't add indexes directly on `JSON` columns anyway.

Reply:
> We already handle the non-JSON to JSON case above. 

You're right! We changes the `case` ordering and now we don't need this check. Fixed: removed three unnecessary checks in total.

Reply:
Yes, we're still left with a few `CONVERT(...)`s yet in the code: for JSONs and for ENUMs. For JSONs it's as you say - not something you can even put in a `primary key` or any `unique key`; for ENUMs it's more complex. I'll take it to another PR.